substrate/frame/contracts/src/exec.rs

```diff
@@ -1252,6 +1280,11 @@ where
 		// Protection is on a per call basis.
 		self.top_frame_mut().allows_reentry = true;
 
+		if set_frame_read_only {
+			// Revert to the previous setting, if it was changed.
+			self.top_frame_mut().read_only = false;
+		}
```

Comment:
I think setting the `read_only` in the parent frame makes it overly complex. We should just have `read_only` as an argument to `push_frame` and do the inheritance here.

Then we don't need to care about resetting it.

Reply:
done